rest_framework/routers.py

```diff
@@ -231,7 +231,7 @@ def get_method_map(self, viewset, method_map):
         """
         bound_methods = {}
         for method, action in method_map.items():
-            if hasattr(viewset, action):
+            if getattr(viewset, action, None) is not None:
```

Comment:
can you please demonstrate a test case which you are describing to do by this change?

Reply:
@auvipy yes, but already at Monday. I'm going to add modelviewset but disable partial_update and patch method.

Reply:
@auvipy I have add simple test for this behaivor. It's may be fix for old conversation #2979 and #3081.